comfy/sample.py

```diff
@@ -0,0 +1,65 @@
+import torch
+import comfy.model_management
+
+
+def prepare_noise(latent, seed):
+    """creates random noise given a LATENT and a seed"""
+    latent_image = latent["samples"]
+    batch_index = 0
+    if "batch_index" in latent:
+        batch_index = latent["batch_index"]
```

Comment:
I feel like the stuff in the comfy folder shouldn't know about the internals of the latent dict. The contents should be passed as params to the function.

Reply:
Tested on inpainting, control nets, selected batch index and gligen. All features seem to be working as intended.